### PR TITLE
support zcash and bitcoin cryptocurrency proofs

### PR DESCRIPTION
**Pull request by maxtaco**
- rename BTC -> Cryptocurrency internally
- new protocol with new name
- support one of each proof at the same time, so slightly change the ID table semantics for this
- rename `keybase btc` -> `keybase currency`
- testing for interaction of BTC and ZCash

### CODE REVIEW

File: go/service/cryptocurrency.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+func NewCryptocurrencyHandler(xp rpc.Transporter, g *libkb.GlobalContext) *CryptocurrencyHandler {
+	return &CryptocurrencyHandler{
+		BaseHandler:  NewBaseHandler(xp),
+		Contextified: libkb.NewContextified(g),
+	}
+}
+
+// BTC creates a BTCEngine and runs it.
```
Comment (patrickxb) [inline go/service/cryptocurrency.go]: old comment?

File: go/service/cryptocurrency.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+func (h *CryptocurrencyHandler) RegisterAddress(_ context.Context, arg keybase1.RegisterAddressArg) error {
+	ctx := engine.Context{
+		LogUI:     h.getLogUI(arg.SessionID),
+		SecretUI:  h.getSecretUI(arg.SessionID, h.G()),
+		SessionID: arg.SessionID,
```
Comment (patrickxb) [inline go/service/cryptocurrency.go]: if you're here, it wouldn't hurt to add `NetContext: nctx`, and replace `_` in params with `nctx`.  Not needed obviously but

**Comment by maxtaco** [comment]
sure thing, will fix!

**Pull request merged**